daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -309,131 +302,103 @@ private[lf] object Anf {
           }
         // It's also safe to perform ANF for applications of a single argument.
         if (safeFunc || args.size == 1) {
```

Comment:
```suggestion
        if (safeFunc || args.lengthCompare(1) == 0) {
```
or 
```suggestion
        if (safeFunc || args.nonEmpty && args.tail.isEmpty) {
```

have a constant complexity instead of linear.